src/components/utils/popover/popover-inline.ts

```diff
@@ -0,0 +1,159 @@
+        class: css.popoverInline,
+      },
+      {
+        [PopoverItemType.Default]: {
+          hint: {
+            position: 'top',
+            alignment: 'center',
+            enabled: isHintEnabled,
+          },
+        },
+        [PopoverItemType.Html]: {
+          hint: {
+            position: 'top',
+            alignment: 'center',
+            enabled: isHintEnabled,
+          },
+        },
+      }
+    );
+
+    /**
+     * If active popover item has children, show them.
+     * This is needed to display link url text (which is displayed as a nested popover content)
+     * once you select <a> tag content in text
+     */
+    this.items
+      .forEach((item) => {
+        if (!(item instanceof PopoverItemDefault) && !(item instanceof PopoverItemHtml)) {
+          return;
+        }
+
+        if (item.hasChildren && item.isActive) {
+          this.showNestedItems(item);
+        }
+      });
+  }
+
+  /**
+   * Returns visible element offset top
+   */
+  public get offsetLeft(): number {
+    if (this.nodes.popoverContainer === null) {
+      return 0;
+    }
+
+    return this.nodes.popoverContainer.offsetLeft;
+  }
+
+  /**
+   * Open popover
+   */
+  public override show(): void {
+    /**
+     * If this is not a nested popover, set CSS variable with width of the popover
+     */
+    if (this.nestingLevel === 0) {
+      this.nodes.popover.style.setProperty(
+        CSSVariables.InlinePopoverWidth,
+        this.renderParams.size.width + 'px'
+      );
+    }
+    super.show();
+  }
+
+  /**
+   * Disable hover event handling
+   */
+  protected override handleHover(): void {
+    // Do nothing
+  }
+
+  /**
+   * Sets CSS variable with position of item near which nested popover should be displayed.
+   * Is used for correct positioning of the nested popover
+   *
+   * @param nestedPopoverEl - nested popover element
+   * @param item – item near which nested popover should be displayed
+   */
+  protected override setTriggerItemPositionProperty(
+    nestedPopoverEl: HTMLElement,
+    item: PopoverItemDefault
+  ): void {
+    const itemEl = item.getElement();
+    const itemOffsetLeft = itemEl ? itemEl.offsetLeft : 0;
+    const totalLeftOffset = this.offsetLeft + itemOffsetLeft;
+
+    nestedPopoverEl.style.setProperty(
+      CSSVariables.TriggerItemLeft,
+      totalLeftOffset + 'px'
+    );
+  }
+
+  /**
+   * Handles displaying nested items for the item.
+   * Overriding in order to add toggling behaviour
+   *
+   * @param item – item to toggle nested popover for
+   */
+  protected override showNestedItems(item: PopoverItemDefault | PopoverItemHtml): void {
+    if (this.nestedPopoverTriggerItem === item) {
+      this.nestedPopoverTriggerItem = null;
+      this.destroyNestedPopoverIfExists();
+
+      return;
+    }
+
+    this.nestedPopoverTriggerItem = item;
+    super.showNestedItems(item);
+  }
+
+  /**
+   * Overrides default item click handling to handle nested popover closing correctly
```

Comment:
describe what does it mean "closing correctly"

Reply:
fixed comment